src/guardian/staking-module-guard/staking-module-guard.service.ts

```diff
@@ -433,7 +436,15 @@ export class StakingModuleGuardService {
   ): boolean {
     if (!firstState || !secondState) return false;
     if (firstState.depositRoot !== secondState.depositRoot) return false;
+
+    // Check if the nonce values are different. A difference in nonce implies a state change.
     if (firstState.nonce !== secondState.nonce) return false;
```

Comment:
Do we still need this check? Seems block_hash check is enough.

Reply:
you are right , we dont need it. but file you check is outdated